File: java-service/src/main/java/com/edu/platform/model/LiveSession.java
package com.edu.platform.model;

import com.edu.platform.model.enums.SessionStatus;
import jakarta.persistence.*;
import lombok.*;

import java.time.Instant;
import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "live_sessions", schema = "academic")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class LiveSession {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    // ==================== Relationships ====================
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "lesson_topic_id")
    private LessonTopic lessonTopic;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "teacher_id", nullable = false)
    private User teacher;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "subject_id", nullable = false)
    private Subject subject;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "class_id")
    private ClassEntity classEntity;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "term_id")
    private Term term;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "created_by")
    private User createdBy;

    @OneToMany(mappedBy = "liveSession", cascade = CascadeType.ALL, orphanRemoval = true)
    @Builder.Default
    private Set<SessionRecording> recordings = new HashSet<>();

    @OneToMany(mappedBy = "liveSession", cascade = CascadeType.ALL, orphanRemoval = true)
    @Builder.Default
    private Set<SessionAttendance> attendances = new HashSet<>();

    // ==================== Zoom Meeting Details ====================
    
    @Column(name = "zoom_meeting_id", unique = true, nullable = false, length = 50)
    private String zoomMeetingId;

    @Column(name = "zoom_meeting_uuid", length = 100)
    private String zoomMeetingUuid;

    @Column(name = "meeting_password", length = 50)
    private String meetingPassword;

    @Column(name = "join_url", nullable = false, columnDefinition = "TEXT")
    private String joinUrl;

    @Column(name = "start_url", nullable = false, columnDefinition = "TEXT")
    private String startUrl;

    // ==================== Scheduling ====================
    
    @Column(name = "scheduled_start_time", nullable = false)
    private Instant scheduledStartTime;

    @Column(name = "scheduled_duration_minutes", nullable = false)
    @Builder.Default
    private Integer scheduledDurationMinutes = 60;

    @Column(name = "actual_start_time")
    private Instant actualStartTime;

    @Column(name = "actual_end_time")
    private Instant actualEndTime;

    // ==================== Status & Metadata ====================
    
    @Enumerated(EnumType.STRING)
    @Column(name = "status", nullable = false, length = 20)
    @Builder.Default
    private SessionStatus status = SessionStatus.SCHEDULED;

    @Column(name = "title", nullable = false, length = 200)
    private String title;

    @Column(name = "description", columnDefinition = "TEXT")
    private String description;

    @Column(name = "max_participants")
    @Builder.Default
    private Integer maxParticipants = 100;

    // ==================== Recording References ====================
    
    @Column(name = "has_recording")
    @Builder.Default
    private Boolean hasRecording = false;

    @Column(name = "recording_processed")
    @Builder.Default
    private Boolean recordingProcessed = false;

    // ==================== Audit Fields ====================
    
    @Column(name = "created_at", nullable = false, updatable = false)
    private Instant createdAt;

    @Column(name = "updated_at")
    private Instant updatedAt;

    @PrePersist
    protected void onCreate() {
        this.createdAt = Instant.now();
        this.updatedAt = Instant.now();
    }

    @PreUpdate
    protected void onUpdate() {
        this.updatedAt = Instant.now();
    }
}

File: python-service/app/models/live_session.py
from sqlalchemy import Column, Integer, String, Text, DateTime, Boolean, ForeignKey, Enum as SQLEnum
from sqlalchemy.orm import relationship
from app.models.base import Base
import enum
from datetime import datetime


class SessionStatus(str, enum.Enum):
    SCHEDULED = "SCHEDULED"
    IN_PROGRESS = "IN_PROGRESS"
    COMPLETED = "COMPLETED"
    CANCELLED = "CANCELLED"


class LiveSession(Base):
    """LiveSession model - matches Java LiveSession entity"""
    __tablename__ = "live_sessions"
    __table_args__ = {'schema': 'academic'}

    id = Column(Integer, primary_key=True, index=True)

    # Foreign Keys
    lesson_topic_id = Column(Integer, ForeignKey('academic.lesson_topics.id'))
    teacher_id = Column(Integer, ForeignKey('academic.users.id'), nullable=False)
    subject_id = Column(Integer, ForeignKey('academic.subjects.id'), nullable=False)
    class_id = Column(Integer, ForeignKey('academic.classes.id'))
    term_id = Column(Integer, ForeignKey('academic.terms.id'))
    created_by_id = Column(Integer, ForeignKey('academic.users.id'), name="created_by")

    # Relationships - ONLY @ManyToOne side (no back_populates)
    lesson_topic = relationship("LessonTopic", lazy="select", foreign_keys=[lesson_topic_id])
    subject = relationship("Subject", lazy="select", foreign_keys=[subject_id])
    term = relationship("Term", lazy="select", foreign_keys=[term_id])
    
    # Zoom Meeting Details
    zoom_meeting_id = Column(String(50), unique=True, nullable=False)
    zoom_meeting_uuid = Column(String(100))
    meeting_password = Column(String(50))
    join_url = Column(Text, nullable=False)
    start_url = Column(Text, nullable=False)

    # Scheduling
    scheduled_start_time = Column(DateTime(timezone=True), nullable=False)
    scheduled_duration_minutes = Column(Integer, nullable=False, default=60)
    actual_start_time = Column(DateTime(timezone=True))
    actual_end_time = Column(DateTime(timezone=True))

    # Status & Metadata
    status = Column(SQLEnum(SessionStatus), nullable=False, default=SessionStatus.SCHEDULED)
    title = Column(String(200), nullable=False)
    description = Column(Text)
    max_participants = Column(Integer, default=100)

    # Recording References
    has_recording = Column(Boolean, default=False)
    recording_processed = Column(Boolean, default=False)

    # Audit Fields
    created_at = Column(DateTime(timezone=True), nullable=False, default=datetime.utcnow)
    updated_at = Column(DateTime(timezone=True), default=datetime.utcnow, onupdate=datetime.utcnow)